lib/IntermediateRepresentation/program.h

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2019, NVIDIA CORPORATION.  All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef FORTRAN_INTERMEDIATEREPRESENTATION_PROGRAM_H_
+#define FORTRAN_INTERMEDIATEREPRESENTATION_PROGRAM_H_
+
+#include "common.h"
+#include "procedure.h"
+#include "../evaluate/type.h"
+#include "llvm/ADT/StringMap.h"
+#include "llvm/ADT/StringRef.h"
+#include "llvm/ADT/Twine.h"
+#include <string>
+
+namespace Fortran::IntermediateRepresentation {
+
+struct Procedure;
```

Comment:
If you compile with clang, it'll warn about mismatching `struct`/`class` keywords.  If these forward declarations are resolved by `class` definitions, they should be `class` here as well.

Reply:
Will do.